### DISCUSSION TITLE
How to enable telegram bot?

### DISCUSSION THREAD

**Discussion by fungic** [juftin/camply#305] [Q&A] [answered]
Wonderful project ! Thank you for making this. 

I am trying to set up Telegram notifications.  I created a bot with bot father and provided my API and bot ID (tried with and without the "@") in the "camply configure" routine. 

When I run 
`camply test-notifications --notifications telegram`

I get this error: 
`ConnectionError: {"ok":false,"error_code":400,"description":"Bad Request: chat not found"}`

Is there additional setup needed? Thank you!

**Comment by juftin** [comment, marked as answer]
Hey @fungic I think maybe you're using your "Bot ID" and not the "Chat Id"?

Full disclosure, the Discord functionality was added by the open-source community so I've never used it myself. There's some context here on this PR: https://github.com/juftin/camply/pull/73

Someone even suggested using this to get your chat id using your bot id: https://codesandbox.io/s/get-telegram-chat-id-q3qkk

**Comment by fungic** [comment]
That was it! I was able to get the test notification sent successfully.  Thank you so much for the quick reply @juftin. I really love the work you did that enables people to get out more... I spent about 4 hours last night starting from scratch, downloading Python for the first time, learning about containers, pipx, etc., because I was really wanted this to work.  As a follow up question, I noticed the default setting in 'camply configure' has an email address. Does that mean that if I don't put anything in the email field, notifications will get sent there as well as to Telegram, as I have it setup? - thanks!

**Reply by juftin** [reply]
🙌 okay that is awesome. I'm sure that Python / Docker knowledge will serve you well for a long time!

Great question about the `email` field. I believe the field you're talking about is actually the `EMAIL_FROM_ADDRESS`
https://github.com/juftin/camply/blob/bcfdd8da4873dc10a691b9b399e798a78740a079/docs/examples/example.camply#L17-L24

That field is only used when you ask for `email` notifications. It makes the email notifications look like they're coming from `camply@juftin.com` but under the hood it would use your own email server, and so would need you to supply your own email server credentials if you want to send emails.

When you're running camply from the commandline you can specify multiple notification providers:

```shell
camply \
  campsites \
  --campground 123 \
  --start-date 2023-10-01 \
  --end-date 2023-11-01 \
  --notifications telegram \
  --notifications email
```

Here's what's that would look like if you're using a YAML config:

```yaml
provider: RecreationDotGov
campgrounds: 
  - 123
start_date: 2023-10-01
end_date: 2023-11-01
notifications: 
  - telegram
  - email
```

If you had both `telegram` and `email` specified it would send you notifications to both places -. In your case you're probably just asking for the `telegram` notification provider and so it won't attempt to use any of that email config and will only send notifications to telegram.

**Reply by fungic** [reply]
Very cool. Thanks again stranger - take care!
